File: testProlog/JavaTest/IntStackTest.java
package iProlog;

import static org.junit.Assert.*;

import org.junit.Before;
import org.junit.Test;
import static java.lang.System.out;


public class IntStackTest {
	IntStack s;
	@Before
	public void setUp() throws Exception {
		s = new IntStack();
	}

	@Test
	public void testCreateStack() {
		assertEquals(s.getTop(), -1);
		out.println("s.getTop : " + s.getTop() + "= -1");
	}
	
	@Test
	public void testPushStack() {
		s.push(1);
		assertEquals(s.getTop(), 0);
		out.println("s.getTop : " + s.getTop() + "= 0");
	}
	
	@Test
	public void testPushStackTwice() {
		s.push(1);
		s.push(2);
		assertEquals(s.getTop(), 1);
		out.println("s.getTop : " + s.getTop() + "= 1");
	}
	
	@Test
	public void testPopStack() {
		s.push(12);
		s.push(10);
		int i = s.pop();
		assertEquals(i, 10);
		out.println("s.pop() : " + i + "= 10");
	}
	
	@Test
	public void testSize() {
		s.push(2);
		s.push(4);
		s.push(10);
		assertEquals(s.size(), 3);
		out.println("s.size() : " + s.size() + "= 3");
	}

}

/*
 * Output of unittest
s.getTop : -1= -1
s.size() : 3= 3
s.getTop : 0= 0
s.pop() : 10= 10
s.getTop : 1= 1
 
 **/

File: testProlog/CppTest/IntStackS/Tests/IntStackTest.cpp
//
// Created by Bidbip on 11/12/2017.
//

/*
Garrett Addington

output
s.getTop: -1= -1
s.getTop: 0= 0
s.getTop: 1= 1
s.pop(): 10= 10
s.size(): 3= 3

exact same as the one in java. works

*/

#include <gtest/gtest.h>
#include <gmock/gmock.h>
#include "IntStack.h"


using testing::Eq;

namespace {
    class ClassDeclaration : public testing::Test {
    public:
        IntStack obj;
        ClassDeclaration() {
            obj;
        }
    };
}

TEST_F(ClassDeclaration, testCreateStack) {
    ASSERT_EQ(obj.getTop(), -1);
    std::cout << "s.getTop: " << obj.getTop() << "= -1" << std::endl;
}
TEST_F(ClassDeclaration, testPushStack) {
    obj.push(1);
    ASSERT_EQ(obj.getTop(), 0);
    std::cout << "s.getTop: " << obj.getTop() << "= 0" << std::endl;

}
TEST_F(ClassDeclaration, testPushStackTwice) {
    obj.push(1);
    obj.push(2);
    ASSERT_EQ(obj.getTop(), 1);
    std::cout << "s.getTop: " << obj.getTop() << "= 1" << std::endl;
}
TEST_F(ClassDeclaration, testPopStack) {
    obj.push(12);
    obj.push(10);
    int i = obj.pop();
    ASSERT_EQ(i, 10);
    std::cout << "s.pop(): " << i << "= 10" << std::endl;
}
TEST_F(ClassDeclaration, testStackSize) {
    obj.push(2);
    obj.push(4);
    obj.push(10);
    ASSERT_EQ(obj.size(),3);
    std::cout << "s.size(): " << obj.size() << "= 3" << std::endl;
}


/*
 *
s.getTop: -1= -1
s.getTop: 0= 0
s.getTop: 1= 2
s.pop(): 10= 10
s.size(): 3= 3
 Removed the testting process and left the output to compare with java script
[==========] 5 tests from 1 test case ran. (5 ms total)
[  PASSED  ] 5 tests.
 *
 *
 */
